caluma/caluma_user/models.py

```diff
@@ -24,7 +24,16 @@ def __init__(self, token: str, userinfo: dict = None, introspection: dict = None
         super().__init__()
 
         self.claims, self.claims_source = self._get_claims(userinfo, introspection)
-        self.username = self.claims[settings.OIDC_USERNAME_CLAIM]
+
+        if (
+            self.claims_source == "introspection"
+            and settings.OIDC_CLIENT_AS_USERNAME
+            and settings.OIDC_USERNAME_CLAIM not in self.claims
+        ):
+            self.username = self.claims[settings.OIDC_CLIENT_CLAIM]
+        else:
+            self.username = self.claims[settings.OIDC_USERNAME_CLAIM]
```

Comment:
I prefer the default to be on top, tather than buried inside an `else`.

```suggestion
        self.username = self.claims[settings.OIDC_USERNAME_CLAIM]
        if (
            self.claims_source == "introspection"
            and settings.OIDC_CLIENT_AS_USERNAME
            and settings.OIDC_USERNAME_CLAIM not in self.claims
        ):
            self.username = self.claims[settings.OIDC_CLIENT_CLAIM]
```

Reply:
This way it will crash, as `settings.OIDC_USERNAME_CLAIM in self.claims == False`

But we could invert the `if` statement to achieve the same thing

Reply:
Ah, I see it now :see_no_evil: I've updated my suggestion above, so it wouldn't crash. But you can also do what you suggested in your comment.

Reply:
Then again, the "default" is also the "fallback" in case we *do* have an introspection source, but `settings.OIDC_USERNAME_CLAIM in self.claims == True` so after thinking about it, I'd rather keep it as is